ops/model.py

```diff
@@ -3980,8 +3891,17 @@ def log_split(
 
     def juju_log(self, level: str, message: str) -> None:
         """Pass a log message on to the juju logger."""
+        # We do not trace this call, to avoid an infinite recursion loop.
```

Comment:
This is not quite accurate.

"prevent_recrusion" prevents the recursion, which is out of scope here.

additionally, `juju-log` is exempted from tracing because touching logging in the trace exporter would lead to a feedback loop. IIRC it's both in case of error, but also non-error `logging.debug` that happens to be on the path somewhere.

(we could do the above selectively: trace main thread log but not the tracing transport thread, for example).

```suggestion
        # Note that this hook command is exempted from tracing, because trace exporter touches logging, which may call our custom logging handler, which would trigger this hook command, resulting in a feedback loop.
```

(feel free to hack this up for readability)

Reply:
Ah, yes, I missed updating that comment when I put the anti-recursion system back, sorry. Will fix.